activation/activation.go

```diff
@@ -555,27 +548,41 @@ func (b *Builder) getCommitmentAtx(ctx context.Context) (*types.ATXID, error) {
 
 // PublishActivationTx attempts to publish an atx, it returns an error if an atx cannot be created.
 func (b *Builder) PublishActivationTx(ctx context.Context) error {
-	b.discardChallengeIfStale()
 	logger := b.log.WithContext(ctx)
 
-	if b.challenge != nil {
+	challenge, err := b.loadChallenge()
+	if err != nil {
+		b.log.With().Info("challenge not loaded", log.Err(err))
+	}
+	if challenge != nil && isStale(challenge, b.currentEpoch()) {
```

Comment:
imo `loadChallenge` should already check for staleness and return `nil` if the challenge is stale and needs to be rebuild.